spec/rocket_chat/messages/im_spec.rb

```diff
@@ -0,0 +1,149 @@
+require 'spec_helper'
+
+describe RocketChat::Messages::Im do
+  let(:server) { RocketChat::Server.new(SERVER_URI) }
+  let(:token) { RocketChat::Token.new(authToken: AUTH_TOKEN, userId: USER_ID) }
+  let(:session) { RocketChat::Session.new(server, token) }
+
+  describe '#create' do
+    before do
+      # Stubs for /api/v1/im.create REST API
+      stub_unauthed_request :post, '/api/v1/im.create'
+
+      stub_authed_request(:post, '/api/v1/im.create')
+        .with(
+          body: {
+            username: 'rocket.cat'
+          }.to_json
+        ).to_return(
+          body: {
+            room: {
+              _id: 'Lymsiu4Mn6xjTAan4RtMDEYc28fQ5aHpf4',
+              _updatedAt: '2018-03-26T19:11:50.711Z',
+              t: 'd',
+              msgs: 0,
+              ts: '2018-03-26T19:11:50.711Z',
+              meta: {
+                revision: 0,
+                created: 1522094603745,
+                version: 0
+              },
+              '$loki': 65,
+              usernames: [
+                'rocket.cat',
+                'user.test'
+              ]
+            },
+            success: true
+          }.to_json,
+          status: 200
+        )
+
+      stub_authed_request(:post, '/api/v1/im.create')
+        .with(
+          body: {
+            username: 'non-existent.user'
+          }
+        ).to_return(
+          body: { success: false }.to_json,
+          status: 400
+        )
+    end
+
+    context 'with a valid session' do
+      it 'create a new direct conversation' do
+        im = session.im.create username: 'rocket.cat'
+        expect(im).to be_a RocketChat::Room
+        expect(im.members).to eq %w[rocket.cat user.test]
+      end
+
+      it 'does not send attributes' do
+        expect do
+          session.im.create
+        end.to raise_error RocketChat::StatusError
+      end
+    end
+
+    context 'with an invalid session token' do
+      let(:token) { RocketChat::Token.new(authToken: nil, userId: nil) }
+
+      it 'raises a status error' do
+        expect do
+          session.im.create username: 'rocket.chat'
+        end.to raise_error RocketChat::StatusError, 'You must be logged in to do this.'
+      end
+    end
+
+    context 'when the user no exists' do
+      it 'raises a status error' do
+        expect do
+          session.im.create(username: 'non-existent.user')
+        end.to raise_error RocketChat::StatusError
+      end
+    end
+  end
+
+  describe '#counters' do
+    before do
+      stub_unauthed_request :get, '/api/v1/im.counters'
+
+      stub_authed_request(:get, '/api/v1/im.counters?roomId=rocket.cat&username=')
```

Comment:
you don't appear to be testing any case where the `username` property is provided?